app/src/organisms/ProtocolTimelineScrubber/index.tsx

```diff
@@ -0,0 +1,608 @@
+import * as React from 'react'
+import map from 'lodash/map'
+import reduce from 'lodash/reduce'
+import ViewportList from 'react-viewport-list'
+import {
+  Flex,
+  Box,
+  DIRECTION_COLUMN,
+  SPACING,
+  ALIGN_CENTER,
+  TEXT_TRANSFORM_UPPERCASE,
+  ALIGN_FLEX_END,
+  COLORS,
+  JUSTIFY_SPACE_BETWEEN,
+  ALIGN_STRETCH,
+  TYPOGRAPHY,
+  LegacyStyledText,
+  BaseDeck,
+  PrimaryButton,
+} from '@opentrons/components'
+import { getResultingTimelineFrameFromRunCommands } from '@opentrons/step-generation'
+import {
+  FLEX_ROBOT_TYPE,
+  THERMOCYCLER_MODULE_TYPE,
+  getSimplestDeckConfigForProtocol,
+} from '@opentrons/shared-data'
+import { getCommandTextData } from '../../molecules/Command/utils/getCommandTextData'
+import { CommandText } from '../../molecules/Command'
+import {
+  getAllWellContentsForActiveItem,
+  wellFillFromWellContents,
+} from './utils'
+
+import type {
+  CompletedProtocolAnalysis,
+  LabwareLocation,
+  ProtocolAnalysisOutput,
+  RobotType,
+  RunTimeCommand,
+} from '@opentrons/shared-data'
+import type {
+  LocationLiquidState,
+  ModuleTemporalProperties,
+  PipetteEntity,
+  TimelineFrame,
+} from '@opentrons/step-generation'
+import type { ViewportListRef } from 'react-viewport-list'
+import type { LabwareOnDeck, Module } from '@opentrons/components'
+
+const COMMAND_WIDTH_PX = 240
+
+interface ProtocolTimelineScrubberProps {
+  commands: RunTimeCommand[]
+  analysis: CompletedProtocolAnalysis | ProtocolAnalysisOutput
+  robotType?: RobotType
+}
+
+export const DECK_LAYER_BLOCKLIST = [
+  'calibrationMarkings',
+  'fixedBase',
+  'doorStops',
+  'metalFrame',
+  'removalHandle',
+  'removableDeckOutline',
+  'screwHoles',
+]
+export const VIEWBOX_MIN_X = -84
+export const VIEWBOX_MIN_Y = -10
+export const VIEWBOX_WIDTH = 600
+export const VIEWBOX_HEIGHT = 460
+
+export function ProtocolTimelineScrubber(
+  props: ProtocolTimelineScrubberProps
+): JSX.Element {
+  const { commands, analysis, robotType = FLEX_ROBOT_TYPE } = props
+  const wrapperRef = React.useRef<HTMLDivElement>(null)
+  const commandListRef = React.useRef<ViewportListRef>(null)
+  const [currentCommandIndex, setCurrentCommandIndex] = React.useState<number>(
+    0
+  )
+  const [isPlaying, setIsPlaying] = React.useState<boolean>(true)
+
+  const currentCommandsSlice = commands.slice(0, currentCommandIndex + 1)
+  const { frame, invariantContext } = getResultingTimelineFrameFromRunCommands(
+    currentCommandsSlice
+  )
+  const handlePlayPause = (): void => {
+    setIsPlaying(!isPlaying)
+  }
+
+  React.useEffect(() => {
+    if (isPlaying) {
+      const intervalId = setInterval(() => {
+        setCurrentCommandIndex(prev => {
+          const nextIndex = prev < commands.length - 1 ? prev + 1 : 0
+          commandListRef.current?.scrollToIndex(nextIndex)
+          return nextIndex
+        })
+      }, 1000)
```

Comment:
nit 
make a const for this?

```suggestion
      }, 1000)
```